### PR TITLE
fabtests: Add shm_sighandler_test to runfabtests

### PR DESCRIPTION
**Pull request by wckzhang**
Signed-off-by: William Zhang <wilzhang@amazon.com>

### CODE REVIEW

**Review by shijin-aws — APPROVED** [review]

**Comment by shijin-aws** [comment]
Probably should pend on https://github.com/ofiwg/libfabric/pull/7776 is merged?

File: fabtests/Makefile.am [hunk 1]
```diff
@@ -141,6 +141,7 @@ nobase_dist_config_DATA = \
 	pytest/default/test_shared_ctx.py \
 	pytest/default/test_ubertest.py \
 	pytest/default/test_unexpected_msg.py \
+	pytest/default/test_shm_sighandler.py \
```
Comment (wzamazon) [inline fabtests/Makefile.am:144]: Since this is shm specific test, should we create `shm` directory and put it there?

Comment (shijin-aws) [reply fabtests/Makefile.am:144]: I had that concern too. But the original test code is already created in `regression` which is not prov specific. Not sure whether we should make this test in `fabtests/prov/shm` instead? @j-xiong

Comment (j-xiong) [reply fabtests/Makefile.am:144]: I originally thought about putting the test under prov/shm, but decided to go with a new "regression" category to catch all the future regression tests.  I think either way makes sense. @shefty what's your opinion?

Comment (shefty) [reply fabtests/Makefile.am:144]: I would leave it where it is.  The test itself mostly doesn't care what provider we're calling.  It's checking signal intercept handling, which could apply to any provider.  We could probably rename the test and allow passing in the provider name as an argument.  I would guess this test might also apply to EFA, and could also apply to the link and peer providers that use shm.

Comment (wckzhang) [reply fabtests/Makefile.am:144]: If you think so...efa is also dependent on shm so I thought it'd fit best in the default dir (and it should work on any build system)

**Review by shefty — APPROVED** [review]
Anyone know what it will take for CI to pass?  Changes look fine to me.

**Comment by shijin-aws** [comment]
bot:aws:retest

**Comment by aingerson** [comment]
@shefty I think it needs #7777 to pass

**Comment by shefty** [comment]
Do we need to rebase on #7777?

**Comment by aingerson** [comment]
CI failed again but for different reasons. I'll look into this and see if I can figure out why...

**Comment by aingerson** [comment]
See #7801 and rebase once merged to fix the CI issues

**Comment by shijin-aws** [comment]
AWS CI failure 

```
command: ssh -n -o StrictHostKeyChecking=no -o ConnectTimeout=30 -o BatchMode=yes 172.31.52.47 timeout 120 /home/ec2-user/PortaFiducia/libraries/libfabric/pr7778-dso/install/fabtests/bin/shm_sighandler_test -p efa
stdout: 
fi_getinfo(): common/shared.c:903, ret=-61 (No data available)
Failed to initialize shm provider
Fail: child killed by SIGKILL or exited with error
```

This is the efa DSO build, which means shm is not available by nature, we will adjust our CI to mitigate this issue.

**Comment by shijin-aws** [comment]
This PR should pass the CI after rebasing on https://github.com/ofiwg/libfabric/pull/7802

**Comment by wckzhang** [comment]
I don't think the remaining CI failures are related

**Comment by shijin-aws** [comment]
@shefty @j-xiong could you help disclose what's the error in Jenkins CI?

The error in appveyor/pr is 

```
- name:   multi_ep -e rdm -v -p sockets
  timestamp: Mon 06/06/2022 19:50:44.18
  result: Fail
  time:   185
  server_cmd: C:\projects\libfabric\fabtests\x64\Debug-v141\multi_ep -e rdm -v -p sockets  -s 127.0.0.1
  server_stdout: 
Creating 3 EPs
Send to all 3 remote EPs
Wait for all messages from peer
Data check OK
PASSED multi ep
```

which seems to be unrelated to this PR.

**Comment by j-xiong** [comment]
There are 4 failures in total. One of them is this test case:
```
11:37:05  - name:   fi_av_xfer -e rdm -p "udp"
11:37:05    timestamp: 20220606-183658+0000
11:37:05    result: Fail
11:37:05    time:   115
11:37:05    server_cmd: /home/cstbuild/ofi-Install/libfabric/ofi_libfabric/PR-7778/4/reg/bin/fi_av_xfer -e rdm -p "udp"   -s ci19-eth2
11:37:05    server_stdout: |
11:37:05      AV address removal: FAIL
11:37:05      bind: Address already in use
11:37:05    client_cmd: /home/cstbuild/ofi-Install/libfabric/ofi_libfabric/PR-7778/4/reg/bin/fi_av_xfer -e rdm -p "udp"   -s ci20-eth2 ci19-eth2
11:37:05    client_stdout: |
11:37:05      [error] fabtests:common/shared.c:2610: cq_readerr 111 (Connection refused), provider errno: 0 (Success)
11:37:05      ft_tx(): functional/av_xfer.c:57, ret=111 (Connection refused)
11:37:05      AV address removal: FAIL
......
11:37:10  - name:   sighandler_test -p "psm3"
11:37:10    timestamp: 20220606-183658+0000
11:37:10    result: Fail
11:37:10    time:   1
11:37:10    server_cmd: /home/cstbuild/ofi-Install/libfabric/ofi_libfabric/PR-7778/4/reg/bin/sighandler_test -p "psm3"
11:37:10    server_stdout: |
11:37:10      Fail: child killed by SIGKILL or exited with error
11:37:10  - name:   fi_av_xfer -e rdm -p "psm3"
11:37:10    timestamp: 20220606-183702+0000
11:37:10    result: Fail
11:37:10    time:   4
11:37:10    server_cmd: /home/cstbuild/ofi-Install/libfabric/ofi_libfabric/PR-7778/4/reg/bin/fi_av_xfer -e rdm -p "psm3"   -s ci1-ib0
11:37:10    server_stdout: |
11:37:10      AV address removal: FAIL
11:37:10      bind: Address already in use
11:37:10    client_cmd: /home/cstbuild/ofi-Install/libfabric/ofi_libfabric/PR-7778/4/reg/bin/fi_av_xfer -e rdm -p "psm3"   -s ci2-ib0 ci1-ib0
11:37:10    client_stdout: |
11:37:10      fi_getinfo(): common/shared.c:903, ret=-61 (No data available)
11:37:10      AV address removal: FAIL
11:37:10      AV re-insertion address: FAIL
11:37:10      fi_getinfo(): common/shared.c:903, ret=-61 (No data available)
......
11:37:00  - name:   fi_rma_bw -e rdm -o writedata -I 5 -p "shm"
11:37:00    timestamp: 20220606-183658+0000
11:37:00    result: Fail
11:37:00    time:   9
11:37:00    server_cmd:  /home/cstbuild/ofi-Install/libfabric/ofi_libfabric/PR-7778/4/reg/bin/fi_rma_bw -e rdm -o writedata -I 5 -p "shm"  -D ze -i 0  -s ci9
11:37:00    server_stdout: |
11:37:00      [error] fabtests:common/shared.c:2610: cq_readerr 22 (Invalid argument), provider errno: -22 (Invalid argument)
11:37:00    client_cmd:  /home/cstbuild/ofi-Install/libfabric/ofi_libfabric/PR-7778/4/reg/bin/fi_rma_bw -e rdm -o writedata -I 5 -p "shm"  -D ze -i 0  -s ci9 ci9
11:37:00    client_stdout: |
11:37:00      [error] fabtests:common/shared.c:2610: cq_readerr 22 (Invalid argument), provider errno: -22 (Invalid argument)
```

**Comment by shijin-aws** [comment]
@j-xiong we don't have a infra to test psm3, could you check what's the error when running `sighandler_test -p "psm3"` ?

**Comment by j-xiong** [comment]
Individual run of "sighandler_test -p psm3" passed just fine. The error could be temporary.

**Comment by shijin-aws** [comment]
@j-xiong Thank you! Since the error should be transient I will merge this PR then.

**Pull request merged**